server/etcdserver/apply.go

```diff
@@ -823,6 +823,9 @@ func (a *applierV3backend) AuthStatus() (*pb.AuthStatusResponse, error) {
 }
 
 func (a *applierV3backend) Authenticate(r *pb.InternalAuthenticateRequest) (*pb.AuthenticateResponse, error) {
+	if a.s.ctx == nil {
```

Comment:
This will for sure fix the fuzz issue, however are we sure returning error here is correct?
Could you check applierV3backend code and verify why ctx is nil here?

Reply:
In this particular case, the server had not ben started and a `ctx` had not been assigned here:

https://github.com/etcd-io/etcd/blob/bdb13e2e12d44a6eb83d35f7867e3c6b9385655b/server/etcdserver/server.go#L554

Starting the server fixed the OSS-fuzz issue, but there is still a crash in `Authenticate` that may cause problems in the future.

Reply:
Is it worth fixing case by case if it's not a realistic issue ? Or there is known sequence of actions that leads to a.s.ctx being null that can harm user. 

The code is running with assumptions about environment, in particular that `a`, `a.s`, `a.s.ctx` are set. 
If we would like to pedantic about checking such assumption probably we should have one method: 

`err := a.validate()` that underneath calls `a.s.validate()` that ensure that ensures that the ctx is set (and other variables). But calling it everywhere could have performance impact. 

IMHO checking it in newApplierV3Backend (https://github.com/etcd-io/etcd/blob/2ea702c852aab612cc1666a68fb108775a3b9b97/server/etcdserver/apply.go#L111-L112) might be sufficient to guarantee that constraint.